src/matlab2tikz.m

```diff
@@ -3543,60 +3543,144 @@ function warnFacetedInterp(m2t)
     % Prescale marker size for octave
     if strcmpi(getEnvironment(), 'Octave')
-        sData = sData.^2/2;
+        dataInfo.Size = dataInfo.Size.^2/2;
     end
-    sData = translateMarkerSize(m2t, markerInfo.style, sqrt(sData)/2);
+    dataInfo.Size = translateMarkerSize(m2t, markerInfo.style, sqrt(dataInfo.Size)/2);
 
     drawOptions = opts_new();
-    if length(cData) == 3
-        [m2t, drawOptions] = getScatterOptsOneColor(m2t, h, drawOptions, ...
-                                                markerInfo, cData, sData, ...
-                                                constMarkerkSize);
-    elseif size(cData,2) == 3
-        drawOptions = getScatterOptsRGB(m2t, drawOptions);
+    
+    %% Determine if we are drawing an actual scatter plot
+    hasDifferentSizes  = numel(dataInfo.Size) ~= 1;
+    hasDifferentColors = numel(dataInfo.C)    ~= 3;
+    isaScatter         = hasDifferentSizes || hasDifferentColors;
+    if isaScatter
+        drawOptions = opts_add(drawOptions, 'scatter');
+    end
+    %TODO: we need to set the scatter source
+    drawOptions = opts_add(drawOptions, 'only marks');
+    drawOptions = opts_add(drawOptions, 'mark', markerInfo.tikz);
+
+    if length(dataInfo.C) == 3
+        % gets options specific to scatter plots with a single color
+        % No special treatment for the colors or markers are needed.
+        % All markers have the same color.
+        [m2t, xcolor, hasFaceColor] = getColorOfMarkers(m2t, h, 'MarkerFaceColor', dataInfo.C);
+        [m2t, ecolor, hasEdgeColor] = getColorOfMarkers(m2t, h, 'MarkerEdgeColor', dataInfo.C);
+
+        if length(dataInfo.Size) == 1;
+            drawOptions = opts_addSubOpts(drawOptions, 'mark options', ...
+                                       markerInfo.options);
+            drawOptions = opts_add(drawOptions, 'mark size', ...
+                sprintf('%.4fpt', dataInfo.Size)); % FIXME: investigate whether to use `m2t.ff`
+            if hasEdgeColor
+                drawOptions = opts_add(drawOptions, 'draw', ecolor);
+            else
+                drawOptions = opts_add(drawOptions, 'color', xcolor); %TODO: why do we even need this one?
+            end
+            if hasFaceColor
+                drawOptions = opts_add(drawOptions, 'fill', xcolor);
+            end
+        else % if changing marker size but same color on all marks
+            markerOptions = opts_new();
+            markerOptions = opts_addSubOpts(markerOptions, 'mark options', ...
+                                         markerInfo.options);
+            if hasEdgeColor
+                markerOptions = opts_add(markerOptions, 'draw', ecolor);
+            else
+                markerOptions = opts_add(markerOptions, 'draw', xcolor);
+            end
+            if hasFaceColor
+                markerOptions = opts_add(markerOptions, 'fill', xcolor);
+            end
+            % for changing marker size, the 'scatter' option has to be added
```

Comment:
It could very well be, but if you look in the old code (below), such a check was already present. 

Personally, I'd say that further clean up should happen in #925 to get this PR merged,  then I have a few days to carry out changes to my thesis and hopefully during the weekend, I can address the other PRs before a 1.1 release.